### ISSUE TITLE
undefined method `transform_keys' for #<ActionDispatch::Request::Session:0x5565f9f0d048 not yet loaded>

### ISSUE THREAD

**Issue by Doltair** [mperham/sidekiq#4452]
Ruby version: ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-linux-musl]
Sidekiq / Pro / Enterprise version(s): 6.0.4
Please include your initializer and any error message with the full backtrace.
   def stringify_keys(other)
          other.transform_keys(&:to_s)
        end
      end

```rack (2.1.1) lib/rack/session/abstract/id.rb:212:in `stringify_keys'
rack (2.1.1) lib/rack/session/abstract/id.rb:148:in `update'
rack (2.1.1) lib/rack/session/abstract/id.rb:317:in `prepare_session'
rack (2.1.1) lib/rack/session/abstract/id.rb:276:in `context'
rack (2.1.1) lib/rack/session/abstract/id.rb:271:in `call'
rack (2.1.1) lib/rack/urlmap.rb:77:in `block in call'
rack (2.1.1) lib/rack/urlmap.rb:61:in `each'
rack (2.1.1) lib/rack/urlmap.rb:61:in `call'
rack (2.1.1) lib/rack/builder.rb:176:in `call'
sidekiq (6.0.4) lib/sidekiq/web.rb:104:in `call'
sidekiq (6.0.4) lib/sidekiq/web.rb:109:in `call'
actionpack (5.2.4.1) lib/action_dispatch/routing/mapper.rb:19:in `block in <class:Constraints>'
actionpack (5.2.4.1) lib/action_dispatch/routing/mapper.rb:48:in `serve'
actionpack (5.2.4.1) lib/action_dispatch/journey/router.rb:52:in `block in serve'
actionpack (5.2.4.1) lib/action_dispatch/journey/router.rb:35:in `each'
actionpack (5.2.4.1) lib/action_dispatch/journey/router.rb:35:in `serve'
actionpack (5.2.4.1) lib/action_dispatch/routing/route_set.rb:840:in `call'
rack-attack (6.2.2) lib/rack/attack.rb:156:in `call'
rack (2.1.1) lib/rack/session/abstract/id.rb:277:in `context'
rack (2.1.1) lib/rack/session/abstract/id.rb:271:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/cookies.rb:670:in `call'
rack-attack (6.2.2) lib/rack/attack.rb:156:in `call'
rack (2.1.1) lib/rack/etag.rb:27:in `call'
rack (2.1.1) lib/rack/conditional_get.rb:27:in `call'
rack (2.1.1) lib/rack/head.rb:14:in `call'
activerecord (5.2.4.1) lib/active_record/migration.rb:559:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
activesupport (5.2.4.1) lib/active_support/callbacks.rb:98:in `run_callbacks'
actionpack (5.2.4.1) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/debug_exceptions.rb:61:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
lograge (0.11.2) lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
railties (5.2.4.1) lib/rails/rack/logger.rb:26:in `block in call'
logstash-logger (0.26.1) lib/logstash-logger/tagged_logging.rb:5:in `block in tagged'
logstash-logger (0.26.1) lib/logstash-logger/tagged_logging.rb:16:in `tagged'
logstash-logger (0.26.1) lib/logstash-logger/tagged_logging.rb:5:in `tagged'
railties (5.2.4.1) lib/rails/rack/logger.rb:26:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
request_store (1.5.0) lib/request_store/middleware.rb:19:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/request_id.rb:27:in `call'
rack (2.1.1) lib/rack/runtime.rb:24:in `call'
activesupport (5.2.4.1) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (5.2.4.1) lib/action_dispatch/middleware/static.rb:127:in `call'
rack (2.1.1) lib/rack/sendfile.rb:113:in `call'
rack-cors (1.1.1) lib/rack/cors.rb:100:in `call'
railties (5.2.4.1) lib/rails/engine.rb:524:in `call'
rack-attack (6.2.2) lib/rack/attack.rb:170:in `call'
rack (2.1.1) lib/rack/tempfile_reaper.rb:17:in `call'
rack (2.1.1) lib/rack/lint.rb:51:in `_call'
rack (2.1.1) lib/rack/lint.rb:39:in `call'
rack (2.1.1) lib/rack/show_exceptions.rb:25:in `call'
rack (2.1.1) lib/rack/common_logger.rb:35:in `call'
rack (2.1.1) lib/rack/chunked.rb:73:in `call'
rack (2.1.1) lib/rack/content_length.rb:17:in `call'
unicorn (5.5.2) lib/unicorn/http_server.rb:605:in `process_client'
unicorn (5.5.2) lib/unicorn/http_server.rb:700:in `worker_loop'
unicorn (5.5.2) lib/unicorn/http_server.rb:548:in `spawn_missing_workers'
unicorn (5.5.2) lib/unicorn/http_server.rb:144:in `start'
unicorn (5.5.2) bin/unicorn:128:in `<top (required)>'
/usr/local/bundle/bin/unicorn:23:in `load'
/usr/local/bundle/bin/unicorn:23:in `<top (required)>'
bundler (2.0.2) lib/bundler/cli/exec.rb:74:in `load'
bundler (2.0.2) lib/bundler/cli/exec.rb:74:in `kernel_load'
bundler (2.0.2) lib/bundler/cli/exec.rb:28:in `run'
bundler (2.0.2) lib/bundler/cli.rb:465:in `exec'
bundler (2.0.2) lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
bundler (2.0.2) lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
bundler (2.0.2) lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
bundler (2.0.2) lib/bundler/cli.rb:27:in `dispatch'
bundler (2.0.2) lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
bundler (2.0.2) lib/bundler/cli.rb:18:in `start'
bundler (2.0.2) exe/bundle:30:in `block in <top (required)>'
bundler (2.0.2) lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
bundler (2.0.2) exe/bundle:22:in `<top (required)>'
/usr/local/bundle/bin/bundle:23:in `load'
/usr/local/bundle/bin/bundle:23:in `<main>'
```

**Comment by Doltair** [comment]
Its failing if I enable these in application.rb
```
# Cookies
    config.middleware.use ActionDispatch::Cookies
    config.middleware.use ActionDispatch::Session::CookieStore
```

**Comment by mperham** [comment]
Please look at recent closed issues.

**Comment by mperham** [comment]
https://github.com/mperham/sidekiq/issues?utf8=✓&q=transform_keys

**Comment by Doltair** [comment]
Gotcha Thank you @mperham. Solved by updating rack ~> 2.1.2
